File: client/src/model/Cnab.ts
interface OperationType {
  id: number;
  description: string;
  type: string;
}

export interface Cnab {
  id: null;
  operationType: OperationType;
  date: string;
  documentId: string;
  value: number;
  cardNumber: string;
  storeOwner: string;
  storeName: string;
}

File: server/src/main/java/com/alexandrecampos/devchallenge/model/Cnab.java
package com.alexandrecampos.devchallenge.model;

import lombok.Data;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "cnab", schema = "core")
public class Cnab {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "operation_type_id")
    private Integer operationTypeId;

    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "operation_type_id", referencedColumnName = "id", insertable = false, updatable = false)
    private OperationType operationType;

    private LocalDateTime date;

    @Column(name = "document_id")
    private String documentId;

    private String value;

    @Column(name = "card_number")
    private String cardNumber;

    @Column(name = "store_owner")
    private String storeOwner;

    @Column(name = "store_name")
    private String storeName;
}
